### ISSUE TITLE
v2.15 (Windows): Invalid ethash share found even before DAG is built

### ISSUE THREAD

**Issue by UselessGuru** [sp-hash/TeamBlackMiner#431]
00:00:03 [2023-12-15 14:03:48.631] GPU0    580S     0.00 H/W      0.00 H/s      0.00 H/s      0/0/**1**        100.00%

```
00:00:00 [2023-12-15 14:00:42.299]
00:00:00 [2023-12-15 14:00:42.457] OpenCL driver version: 546.1
00:00:00 [2023-12-15 14:00:42.492] Cuda driver version: 12.3
00:00:00 [2023-12-15 14:00:42.492] Cuda runtime API version: 12.0
00:00:00 [2023-12-15 14:00:42.492]
00:00:00 [2023-12-15 14:00:42.492]
00:00:00 [2023-12-15 14:00:42.492]
00:00:00 [2023-12-15 14:00:42.492] Welcome to Team Black Miner 2.15
00:00:00 [2023-12-15 14:00:42.492]
00:00:00 [2023-12-15 14:00:42.492]
00:00:00 [2023-12-15 14:00:42.492] GPU0 Ellesmere 8GB added to miner as OpenCL device
00:00:00 [2023-12-15 14:00:42.493] Trying to read verthash data file from disk
00:00:00 [2023-12-15 14:00:42.494] Api server started at 127.0.0.1:4004
00:00:00 [2023-12-15 14:00:42.969] Verthash data file has been loaded succesfully
00:00:00 [2023-12-15 14:00:48.131] Verthash data file has been verified succesfully
00:00:00 [2023-12-15 14:00:48.131] Trying verthash.eu.mine.zergpool.com:4534 (verthash)
00:00:00 [2023-12-15 14:00:48.131] Trying eu.mining.prohashing.com:3339 (ethash)
00:00:00 [2023-12-15 14:00:48.132] GPU0 Intensity set to 256
00:00:00 [2023-12-15 14:00:48.132] GPU0 Started CL work thread
00:00:00 [2023-12-15 14:00:48.306] eu.mining.prohashing.com: Pool responded successfully to subscribe
00:00:00 [2023-12-15 14:00:48.306] eu.mining.prohashing.com: Difficulty is set to 0.500000
00:00:00 [2023-12-15 14:00:48.399] eu.mining.prohashing.com: Pool responded successfully to authorize
00:00:00 [2023-12-15 14:00:48.445] GPU0 Load verthash data size 1283457024
00:00:00 [2023-12-15 14:00:48.599] verthash.eu.mine.zergpool.com: Pool responded successfully to subscribe
00:00:00 [2023-12-15 14:00:48.898] verthash.eu.mine.zergpool.com: Pool responded successfully to extranonce subscribe
00:00:00 [2023-12-15 14:00:48.930] verthash.eu.mine.zergpool.com: Pool responded successfully to authorize
00:00:00 [2023-12-15 14:00:48.946] verthash.eu.mine.zergpool.com: Difficulty is set to 2.000000
00:00:00 [2023-12-15 14:00:49.302] GPU0 Verthash data loaded to gpu
00:00:00 [2023-12-15 14:00:49.309] Job  (verthash)
00:00:00 [2023-12-15 14:00:55.062] GPU0 Starting generating ethash dag (Epoch: 616)
00:00:00 [2023-12-15 14:01:21.324] VERTHASH: GPU0 Accepted (1/1/0) Reply: 357ms
00:00:00 [2023-12-15 14:01:23.498] verthash.eu.mine.zergpool.com: Difficulty is set to 1.000000
00:00:00 [2023-12-15 14:01:31.504] Job  (verthash)
00:00:00 [2023-12-15 14:01:44.857] verthash.eu.mine.zergpool.com: Difficulty is set to 0.500000
00:00:01 [2023-12-15 14:01:48.606] mining.prohashing.com (ethash)    PING: 62ms    DIFFICULTY: 0.500    EPOCH: 616
00:00:01 [2023-12-15 14:01:48.606]
00:00:01 [2023-12-15 14:01:48.606] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:01 [2023-12-15 14:01:48.606] GPU0    580S     OpenCL  0       0       0/0     0       0       0       0
00:00:01 [2023-12-15 14:01:48.606]                                                                          0
00:00:01 [2023-12-15 14:01:48.606] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:01 [2023-12-15 14:01:48.606] GPU0    580S     0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:01 [2023-12-15 14:01:48.606]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:01 [2023-12-15 14:01:48.606]
00:00:01 [2023-12-15 14:01:48.606] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:01 [2023-12-15 14:01:48.606]
00:00:01 [2023-12-15 14:01:48.606] eu.mine.zergpool.com (verthash)    PING: 356ms    VARDIFF: 0.500    EPOCH: 0
00:00:01 [2023-12-15 14:01:48.606]
00:00:01 [2023-12-15 14:01:48.606] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:01 [2023-12-15 14:01:48.606] GPU0    580S     OpenCL  0       0       0/0     0       0       0       0
00:00:01 [2023-12-15 14:01:48.606]                                                                          0
00:00:01 [2023-12-15 14:01:48.606] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:01 [2023-12-15 14:01:48.606] GPU0    580S     0.00 H/W      527.78 kH/s   527.65 kH/s   1/0/0        100.00%
00:00:01 [2023-12-15 14:01:48.606]                  0.00 H/W      527.78 kH/s   527.65 kH/s   1/0/0        100.00%
00:00:01 [2023-12-15 14:01:48.606]
00:00:01 [2023-12-15 14:01:48.606] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:01 [2023-12-15 14:01:48.606]
00:00:01 [2023-12-15 14:01:52.582] Job  (verthash)
00:00:01 [2023-12-15 14:02:05.837] verthash.eu.mine.zergpool.com: Difficulty is set to 0.250000
00:00:01 [2023-12-15 14:02:14.271] Job  (verthash)
00:00:01 [2023-12-15 14:02:25.846] VERTHASH: GPU0 Accepted (2/2/0) Reply: 279ms
00:00:01 [2023-12-15 14:02:36.950] VERTHASH: GPU0 Accepted (3/3/0) Reply: 279ms
00:00:01 [2023-12-15 14:02:47.755] VERTHASH: GPU0 Accepted (4/4/0) Reply: 291ms
00:00:02 [2023-12-15 14:02:48.000] verthash.eu.mine.zergpool.com: Difficulty is set to 0.125000
00:00:02 [2023-12-15 14:02:48.608] mining.prohashing.com (ethash)    PING: 67ms    DIFFICULTY: 0.500    EPOCH: 616
00:00:02 [2023-12-15 14:02:48.608]
00:00:02 [2023-12-15 14:02:48.608] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:02 [2023-12-15 14:02:48.608] GPU0    580S     OpenCL  0       0       0/0     0       0       0       0
00:00:02 [2023-12-15 14:02:48.608]                                                                          0
00:00:02 [2023-12-15 14:02:48.608] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:02 [2023-12-15 14:02:48.608] GPU0    580S     0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:02 [2023-12-15 14:02:48.608]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/0        100.00%
00:00:02 [2023-12-15 14:02:48.608]
00:00:02 [2023-12-15 14:02:48.608] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:02 [2023-12-15 14:02:48.608]
00:00:02 [2023-12-15 14:02:48.608] eu.mine.zergpool.com (verthash)    PING: 312ms    VARDIFF: 0.125    EPOCH: 0
00:00:02 [2023-12-15 14:02:48.608]
00:00:02 [2023-12-15 14:02:48.608] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:02 [2023-12-15 14:02:48.608] GPU0    580S     OpenCL  0       0       0/0     0       0       0       0
00:00:02 [2023-12-15 14:02:48.608]                                                                          0
00:00:02 [2023-12-15 14:02:48.608] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:02 [2023-12-15 14:02:48.608] GPU0    580S     0.00 H/W      518.22 kH/s   525.00 kH/s   4/0/0        100.00%
00:00:02 [2023-12-15 14:02:48.608]                  0.00 H/W      518.22 kH/s   525.00 kH/s   4/0/0        100.00%
00:00:02 [2023-12-15 14:02:48.608]
00:00:02 [2023-12-15 14:02:48.608] SHARES PER MINUTE: 3.00                              POOL HASHRATE: 104.86 kH/s
00:00:02 [2023-12-15 14:02:48.608]
00:00:02 [2023-12-15 14:02:58.464] Job  (verthash)
00:00:02 [2023-12-15 14:03:10.266] VERTHASH: GPU0 Accepted (5/5/0) Reply: 106ms
00:00:02 [2023-12-15 14:03:10.281] VERTHASH: GPU0 Accepted (6/6/0) Reply: 121ms
00:00:02 [2023-12-15 14:03:10.312] VERTHASH: GPU0 Accepted (7/7/0) Reply: 152ms
00:00:02 [2023-12-15 14:03:10.327] VERTHASH: GPU0 Accepted (8/8/0) Reply: 167ms
00:00:02 [2023-12-15 14:03:21.091] VERTHASH: GPU0 Accepted (9/9/0) Reply: 126ms
00:00:02 [2023-12-15 14:03:32.530] Job 43bb (verthash) is possible stale
00:00:02 [2023-12-15 14:03:32.530] Job  (verthash)
00:00:02 [2023-12-15 14:03:43.887] VERTHASH: GPU0 Accepted (10/10/0) Reply: 229ms
00:00:02 [2023-12-15 14:03:43.918] VERTHASH: GPU0 Accepted (11/11/0) Reply: 260ms
00:00:02 [2023-12-15 14:03:43.934] VERTHASH: GPU0 Accepted (12/12/0) Reply: 276ms
00:00:03 [2023-12-15 14:03:48.631] mining.prohashing.com (ethash)    PING: 66ms    DIFFICULTY: 0.500    EPOCH: 616
00:00:03 [2023-12-15 14:03:48.631]
00:00:03 [2023-12-15 14:03:48.631] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:03 [2023-12-15 14:03:48.631] GPU0    580S     OpenCL  0       0       0/0     0       0       0       0
00:00:03 [2023-12-15 14:03:48.631]                                                                          0
00:00:03 [2023-12-15 14:03:48.631] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:03 [2023-12-15 14:03:48.631] GPU0    580S     0.00 H/W      0.00 H/s      0.00 H/s      0/0/1        100.00%
00:00:03 [2023-12-15 14:03:48.631]                  0.00 H/W      0.00 H/s      0.00 H/s      0/0/1        100.00%
00:00:03 [2023-12-15 14:03:48.631]
00:00:03 [2023-12-15 14:03:48.631] SHARES PER MINUTE: 0.00                              POOL HASHRATE: 0.00 H/s
00:00:03 [2023-12-15 14:03:48.631]
00:00:03 [2023-12-15 14:03:48.631] eu.mine.zergpool.com (verthash)    PING: 230ms    VARDIFF: 0.125    EPOCH: 0
00:00:03 [2023-12-15 14:03:48.631]
00:00:03 [2023-12-15 14:03:48.631] ID      BOARD    TYPE    KERN    XINT    TEMP    FAN     CORE    MEM     WATT
00:00:03 [2023-12-15 14:03:48.631] GPU0    580S     OpenCL  0       0       0/0     0       0       0       0
00:00:03 [2023-12-15 14:03:48.631]                                                                          0
00:00:03 [2023-12-15 14:03:48.631] ID      BOARD    HASHRATE/W    HASHRATE      AVERAGE       SHARES       RATE
00:00:03 [2023-12-15 14:03:48.631] GPU0    580S     0.00 H/W      510.86 kH/s   521.39 kH/s   12/0/0       100.00%
00:00:03 [2023-12-15 14:03:48.632]                  0.00 H/W      510.86 kH/s   521.39 kH/s   12/0/0       100.00%
00:00:03 [2023-12-15 14:03:48.632]
00:00:03 [2023-12-15 14:03:48.632] SHARES PER MINUTE: 8.00                              POOL HASHRATE: 279.62 kH/s
00:00:03 [2023-12-15 14:03:48.632]
00:00:03 [2023-12-15 14:03:54.451] Job 43bc (verthash) is possible stale
00:00:03 [2023-12-15 14:03:54.452] Job  (verthash)
00:00:03 [2023-12-15 14:04:05.814] VERTHASH: GPU0 Accepted (13/13/0) Reply: 154ms
00:00:03 [2023-12-15 14:04:17.515] VERTHASH: GPU0 Accepted (14/14/0) Reply: 320ms
00:00:03 [2023-12-15 14:04:17.531] VERTHASH: GPU0 Accepted (15/15/0) Reply: 336ms
00:00:03 [2023-12-15 14:04:17.561] VERTHASH: GPU0 Accepted (16/16/0) Reply: 366ms
00:00:03 [2023-12-15 14:04:28.330] VERTHASH: GPU0 Accepted (17/17/0) Reply: 344ms
```

**Comment by sp-hash** [comment]
this was fixed in v2.16

**Comment by UselessGuru** [comment]
Fixed in build v3
